barretenberg/cpp/src/barretenberg/vm2/tracegen/lib/instruction_spec.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <cstdint>
+#include <unordered_map>
+
+#include "barretenberg/vm2/common/opcodes.hpp"
+
+namespace bb::avm2::tracegen {
+enum class SubtraceSel : uint8_t {
+    ALU,
+    BITWISE,
+    TORADIXBE,
+    POSEIDON2PERM,
+    ECC,
+    DATACOPY,
+    EXECUTION,
+    // Temp
+    OTHERS,
+};
+
+struct SubtraceInfo {
+    SubtraceSel subtrace_selector;
+    uint8_t subtrace_operation_id;
+};
+
+// Non traditional builder
+class RegisterMemInfo {
```

Comment:
This will almost certainly be re-visited if/when we encounter an opcode with an unusual register pattern

Reply:
What does this actually do? I don't understand what it encodes to/from. Could you add a comment?